google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryOptions.java

```diff
@@ -61,6 +64,15 @@ private Builder(BigQueryOptions options) {
     }
 
     @Override
+    public Builder setTransportOptions(TransportOptions transportOptions) {
```

Comment:
There are 7 exact copies of this method, all for ServiceNameOptions classes, all extending ServiceOptions. Since it is already an extension (not implementation), please consider adding HttpServiceOptions class in the hierarcy (or keeping existing one, so HttpTransportOptions would be a completely new class, not a renamed one). setTransportOptions and getTransportOptions are potential candidates to be moved into that class.